### ISSUE TITLE
REPL targets depend on their corresponding target library/executable

### ISSUE THREAD

**Issue by aherrmann** [tweag/rules_haskell#289]
rules_haskell generates REPL targets for Haskell libraries and executables. However, these depend on the library/executable they are for, and will fail to build if that library/executable is in a state where it doesn't build.

Say I have a library `mylib` and want to work on it using the corresponding REPL target. But, I'm in the middle of developing a feature, so somewhere in `mylib` is some type error, or maybe a typed hole. If I issue `bazel run //mylib:mylib-repl` bazel will first try to build `mylib`. That will fail and I cannot open the REPL. I first have to undo any unfinished work and bring `mylib` back to a state where it builds. Then I can build and open the REPL, and then I can bring back my unfinished work and `:reload` in the REPL to continue working. It would be nice if the REPL would build and start straight-away.

**Comment by aherrmann** [comment]
I've tested it and #352 does resolve this issue. Thanks!

**Comment by judah** [comment]
It looks like this behavior has regressed.  If I `bazel build` or `bazel run` the `*-repl` name then Bazel still tries to build the full library first.

For example, in `BUILD`:
```
haskell_library(
    name = "test",
    srcs = ["Test.hs"],
    deps = [hazel_library("base")],
)
```
And with `Test.hs` containing a syntax error:
```
```module Test where

foo =
```

I get: 
```
$ bazel run :test-repl
INFO: Analysed target //:test-repl (0 packages loaded).
INFO: Found 1 target...
ERROR: /Users/judah.jacobson/repos/platform-core/BUILD:47:1: HaskellBuildLibrary //:test failed (Exit 1)

Test.hs:4:1: error:
    parse error (possibly incorrect indentation or mismatched brackets)
Target //:test-repl failed to build
```

**Comment by mboes** [comment]
I suspect this happened in #383.

**Comment by judah** [comment]
I haven't tried to bisect it, but we'te experiencing it at an earlier commit: 979de1592dc820bde86dfcf17165ffa2e8a70055

**Comment by tstellanova** [comment]
@mboes This issue is still impacting our team. Any thoughts on a fix for this regression?

**Comment by mboes** [comment]
@mrkkrp could you have a look?

**Comment by thumphries** [comment]
We have had no problems with this since merging Mark's changes (#441).

**Comment by mboes** [comment]
The fix for this landed in `master` as part of #441. @ghorn are you still having issues even with this? If not let's close.

**Comment by ghorn** [comment]
This is fixed for me.
